src/system-health/tests/test_system_health.py

```diff
@@ -0,0 +1,224 @@
+"""
+    Unit test cases for system health checker. The current test case contains:
+        1. test_user_defined_checker mocks the output of a user defined checker and verify class UserDefinedChecker
+        2. test_service_checker mocks the output of monit service and verify class ServiceChecker
+        3. test_hardware_checker mocks the hardware status data in db and verify class HardwareChecker
+    And there are class that are not covered by unit test. These class will be covered by sonic-mgmt regression test.
+        1. HealthDaemon
+        2. HealthCheckerManager
+        3. Config
+"""
+import os
+import sys
+
+from .mock_connector import MockConnector
+import swsssdk
+
+swsssdk.SonicV2Connector = MockConnector
+from mock import Mock, MagicMock, patch
+
+test_path = os.path.dirname(os.path.abspath(__file__))
+modules_path = os.path.dirname(test_path)
+sys.path.insert(0, modules_path)
+from health_checker.manager import HealthCheckerManager
+from sonic_py_common import device_info
+
+device_info.get_platform = MagicMock(return_value='unittest')
+
+
+def test_user_defined_checker():
+    from health_checker import utils
+    utils.run_command = MagicMock(return_value='')
+
+    from health_checker.user_defined_checker import UserDefinedChecker
+    from health_checker.health_checker import HealthChecker
+    checker = UserDefinedChecker('')
+    checker.check(None)
+    assert checker._info[str(checker)][HealthChecker.INFO_FIELD_OBJECT_STATUS] == HealthChecker.STATUS_NOT_OK
+
+    checker.reset()
+    assert len(checker._info) == 0
+
+    utils.run_command = MagicMock(return_value='\n\n\n')
+    checker.check(None)
+    assert checker._info[str(checker)][HealthChecker.INFO_FIELD_OBJECT_STATUS] == HealthChecker.STATUS_NOT_OK
+
+    valid_output = 'MyCategory\nDevice1:OK\nDevice2:Device2 is broken\n'
+    utils.run_command = MagicMock(return_value=valid_output)
+    checker.check(None)
+    assert 'Device1' in checker._info
+    assert 'Device2' in checker._info
+    assert checker._info['Device1'][HealthChecker.INFO_FIELD_OBJECT_STATUS] == HealthChecker.STATUS_OK
+    assert checker._info['Device2'][HealthChecker.INFO_FIELD_OBJECT_STATUS] == HealthChecker.STATUS_NOT_OK
+
+
+def test_service_checker():
+    from health_checker import utils
+    from health_checker.config import Config
+    from health_checker.health_checker import HealthChecker
+    from health_checker.service_checker import ServiceChecker
```

Comment:
These are imported in a couple test functions. Why not import at the top of the file?